### PR TITLE
Adding ratings

### PR DESCRIPTION
**Pull request by lukehansell-hx**
#### What does this PR do?
Adds the Ratings component to the UI Toolkit

#### What unit or integration tests does this PR have?
None

#### What selenium tests does this PR have?
None

#### How should a developer review this?
Check that the code seems sane and that the documentation describes sufficiently how to use it.
`grunt docs` will build the docs for you to see the example etc.

#### How should this be manually tested?
Run `grunt docs` and check the example.

If you're feeling really daring...

check out the *webapp-template* repo.
run `make init` in the webapp-template.
check out this the *ui-toolkit* and switch to this branch.
in the toolkit `npm link`
in the webapp-template `npm link ui-toolkit`
then, copying the example, add the component to the returned div inside `webapp-template/code/products/views/products_summary/template/index.jsx`
run `make dev` in the webapp-template.
You should now see the component on the page that loads...

#### Any background context you want to provide?
This is another addition to our UI-Toolkit that will bring React to the HX masses... or something like that.

#### What gif best describes how you feel about this work?

---
#### Quality checklist (for PR author to complete BEFORE code review):
- [x] I've checked there is appropriate unit test coverage.
- [x] I've checked there is appropriate selenium test coverage.
- [x] I've included regression tests for any bug fixes.
- [x] I've updated documentation with any changes to procedures.
- [x] I've tested this on mobile Android, Safari, Chrome for any visual changes.
- [x] I've tested this on desktop IE9/10/11, Firefox, Chrome for any visual changes.
- [x] I've checked any significant visual changes with a member of the UX/UI pod.
- [x] I've checked this work against the requirements of the Jira.
- [x] I've added and updated translations for all supported languages
- [x] I've informed those who need to know of any unforeseen impact of this work (database migrations, deployment procedure changes).
- [x] I am ready for this to be code reviewed, merged and tested.
- [x] I've thought about writing a blog post about this

#### IP or Developer Review:
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve run all the tests locally and they pass.
- [x] +1 from me!

#### Software Engineer or Developer review:
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve run all the tests locally and they pass.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] +1 from me!

#### Software Engineer or project guru review:
- [x] I agree with the assumptions made in the quality checklist.
- [x] I’ve witnessed the work behaving as expected.
- [x] I’ve run all the tests locally and they pass.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

### CODE REVIEW

**Comment by jackdcrawford** [comment]
SE/Dev == me

File: docs/examples/Rating.jsx [hunk 1]
```diff
@@ -0,0 +1,4 @@
+var myComponentInstance = (
```
Comment (Joezo) [inline docs/examples/Rating.jsx]: can we stick to the existing style of naming these instances after their type? e.g. `ratingInstance`

File: docs/src/Components.jsx [hunk 2]
```diff
@@ -25,7 +25,21 @@ var Components = React.createClass({
           <p>Need to position your flag? You can position the flag anywhere in a container using <code>top</code> (default), <code>bottom</code>, <code>left</code> or <code>right</code>.</p>
           <p>The container must have a <code>relative:position</code> and a <code>z-index:x</code> applied.</p>
           <CustomComponent codeText={fs.readFileSync(__dirname + '/../examples/FlagPosition.jsx', 'utf8')} />
-        </article>        
+        </article>
+
+        <article>
+          <h3 id="rating">Rating</h3>
+          <p>Ratings can be use to display a series of icons intended to represent a score.</p>
```
Comment (Joezo) [inline docs/src/Components.jsx]: can be *used*

File: docs/src/Components.jsx [hunk 3]
```diff
@@ -25,7 +25,21 @@ var Components = React.createClass({
...
+          <CustomComponent codeText={fs.readFileSync(__dirname + '/../examples/Rating.jsx', 'utf8')} />
+          <h4>Attributes</h4>
+          <ul>
+            <li><code>rating</code> Number - The count of positive icons to display.</li>
+            <li><code>outOf</code> [optional] Number - The total count the rating value is out of.</li>
+            <li><code>blankIcon</code> [optional] Node (any displayable item) - the icon to display for empty ratings.</li>
+          </ul>
+          
+          The <code>outOf</code> and <code>blankIcon</code> attributes are optional and dependent upon each other.
```
Comment (Joezo) [inline docs/src/Components.jsx]: Does this mean you can't have one without the other? If so, why?

File: package.json [hunk 4]
```diff
@@ -30,7 +30,8 @@
     "require-directory": "^2.0.0",
     "requirejs": "~2.1.9",
     "ui-component-button": "git+ssh://git@github.com:holidayextras/ui-component-button.git",
-    "ui-component-flag": "git+ssh://git@github.com:holidayextras/ui-component-flag.git"
+    "ui-component-flag": "git+ssh://git@github.com:holidayextras/ui-component-flag.git",
+    "ui-component-rating": "git://github.com/holidayextras/ui-component-rating"
```
Comment (Joezo) [inline package.json]: Should probably maintain existing reference to github repos, add `+ssh` and `.git`

File: docs/src/Components.jsx (hunk 3 above)
Comment (lukehansell-hx) [inline docs/src/Components.jsx]: you could supply an outOf, but without anything to display nothing will be shown (and vice-versa)

**Comment by Joezo** [comment]
i've taken the ip/dev slottt

**Comment by Joezo** [comment]
It's missing from the menu https://cloudup.com/cH4TGMSk0lX

File: docs/src/Components.jsx (hunk 3 above)
Comment (Joezo) [inline docs/src/Components.jsx]: Makes sense, I wonder if there is a clearer way of wording that in the docs

**Comment by jackdcrawford** [comment]
This looks good :+1:

**Comment by Joezo** [comment]
:+1: left a small comment on making the docs slightly clearer but not a showstopper :)

**Comment by hx-markterry** [comment]
taken

**Comment by hx-markterry** [comment]
ok with me :blue_car:

**Comment by lukehansell-hx** [comment]
@Joezo and @jackdcrawford you've both +1ed but haven't ticked all the boxes... could you quickly go over them please?

**Pull request merged**